react-big-calendar/index.d.ts

```diff
@@ -26,10 +26,10 @@ declare module 'react-big-calendar' {
         step?: number;
         rtl?: boolean;
         eventPropGetter?: (event: Object, start: stringOrDate, end: stringOrDate, isSelected: boolean) => void;
-        titleAccessor?: string;
-        allDayAccessor?: boolean;
-        startAccessor?: stringOrDate;
-        endAccessor?: stringOrDate;
+        titleAccessor?: string | ((row: Object) => string);
+        allDayAccessor?: string | ((row: Object) => boolean);
```

Comment:
This is a good idea, unfortunately it will not work as you would wish and would require consume to do some workarounds which is not optimal for my taste. To use a generic JSX element currently, you must alias a type and interface or extend a base class to a non-generic specialization:

e.g. `class MyTypeBigCalendar extends BigCalendar<MyType> {}`

And I think with stricter type checking like enabled `noImplicit*`, would not have a possibility to opt-out.

There is an existing issue for this already, so I would suggest wait for it to be finalized, before introducing this change. Issue: https://github.com/Microsoft/TypeScript/issues/6395